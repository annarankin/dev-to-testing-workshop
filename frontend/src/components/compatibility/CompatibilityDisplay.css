.CompatibilityDisplay {
  overflow: hidden;
  padding: 8px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
