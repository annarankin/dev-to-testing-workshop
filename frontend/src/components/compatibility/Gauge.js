import React from 'react';
import { compatibilityLevel } from '../../compatibility_calculator'
// import './Gauge.css'

const Gauge = ({ listA, listB }) => {
  return (
    <div className="Gauge">
      <h3>Compatibility Level: {compatibilityLevel(listA, listB)}%</h3>
    </div>
  )
}

export default Gauge
