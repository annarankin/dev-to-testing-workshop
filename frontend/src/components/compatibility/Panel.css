.Panel {
  width: 25%;
  max-width: 300px;
}

.Panel .profile-img {
  width: 100%;
  min-height: 250px;
  background-size: cover;
  background-position: center;
}
