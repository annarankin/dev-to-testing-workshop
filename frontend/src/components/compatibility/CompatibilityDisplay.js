import React from 'react';
import Gauge from './Gauge'
import Panel from './Panel'
import './CompatibilityDisplay.css'

const CompatibilityDisplay = ({ dogeA, dogeB }) => {
  if (!dogeA || !dogeB) { return <div>Loading!</div> }

  return (
    <div className="CompatibilityDisplay">
      <Panel {...dogeA} />
      <Gauge
        listA={dogeA.likes}
        listB={dogeB.likes}
      />
      <Panel {...dogeB} />
    </div>
  )
}

export default CompatibilityDisplay
