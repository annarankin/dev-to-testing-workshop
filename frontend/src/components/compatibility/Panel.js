import React from 'react';
import './Panel.css'

const Panel = ({
  name,
  profile_image,
  caption,
  likes,
}) => {
  const profileImgStyle = {
    backgroundImage: `url('${profile_image}')`
  }

  return (
    <div className="Panel">
      <div className="profile-img" style={profileImgStyle} />

      <table>
        <thead>
          <tr><th>{name}</th></tr>
        </thead>

        <tbody>
          <tr><td>"{caption}"</td></tr>
          <tr><td>Likes: {likes.join(', ')}</td></tr>
        </tbody>
      </table>
    </div>
  )
}

export default Panel
