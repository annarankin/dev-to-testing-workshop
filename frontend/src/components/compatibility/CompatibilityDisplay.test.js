// Provides a wrapper around a React component
// that gives us awesome testing methods!
// http://airbnb.io/enzyme/docs/api/
import { shallow } from 'enzyme';

import React from 'react';
import ReactDOM from 'react-dom';
import CompatibilityDisplay from './CompatibilityDisplay';
import Panel from './Panel';
import Gauge from './Gauge';

describe('when given an object that contains "dogeA" and "dogeB"', () => {
  it('renders a child components', () => {
    const props = { dogeA: true, dogeB: true }
    const componentWrapper = shallow(<CompatibilityDisplay {...props} />)

    expect(componentWrapper.contains([<Panel />, <Gauge />, <Panel />])).toBe(true)
  })

  // Note - the above test is pretty useless compared to a simple snapshot test:
  it('matches snapshot', () => {
    const props = { dogeA: true, dogeB: true }
    const componentWrapper = shallow(<CompatibilityDisplay {...props} />)

    expect(componentWrapper).toMatchSnapshot()
  })
})

describe('with empty props', () => {
  it('displays a loading indicator', () => {
    const componentWrapper = shallow(<CompatibilityDisplay />)

    expect(componentWrapper.text()).toEqual('Loading!')
  })
})

