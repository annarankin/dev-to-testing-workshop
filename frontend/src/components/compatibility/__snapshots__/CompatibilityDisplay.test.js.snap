// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when given an object that contains "dogeA" and "dogeB" matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompatibilityDisplay
    dogeA={true}
    dogeB={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Panel />,
        <Gauge
          listA={undefined}
          listB={undefined}
/>,
        <Panel />,
      ],
      "className": "CompatibilityDisplay",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "listA": undefined,
          "listB": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Panel />,
          <Gauge
            listA={undefined}
            listB={undefined}
/>,
          <Panel />,
        ],
        "className": "CompatibilityDisplay",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "listA": undefined,
            "listB": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
